### ISSUE TITLE
Race condition on shutdown

### ISSUE THREAD

**Issue by SebastianBecker2** [hudec117/Mpv.NET-lib-#28]
**Mpv.NET (lib) Version** current master

**Describe the bug**
When the shutdown event is triggered by MPV, the eventLoopTask gets the event and calls MpvEvents::HandleShutdown().

Here is the first issue. It calls functions.TerminateDestroy(handle) which destroys the handle. And it then calls eventLoop::Stop() which tries to call the Functions.Wakeup(mpvHandle) which fails since the handle is destroyed.

The main issue though is that this is all done in the eventLoopTask. So even if we switch the calls to functions.TerminateDestroy(handle) and eventLoop::Stop() i the MpvEvent::HandleShutdown() function, the Functions.Wakeup(mpvHandle) call is followed by waiting for the eventLookTask to finish. So it is waiting for itself to finish and deadlocking.

**To Reproduce**
```csharp
using (var mpv = new Mpv.NET.API.Mpv(libPath))
{
    mpv.Command("loadfile", filePath);
    mpv.EndFile += (s, a) => mpv.Command("quit");
    while (mpv.EventLoop.IsRunning)
    {
    }
}
```
Though please keep in mind, I'm still quite unsure how I'm supposed to handle the EventLoop. Advice is welcome.

**Expected behavior**
I would expect the EventLoop to stop when the "quit" command is called. Thus leaving the loop and disposing the mpv object when it goes out of the using scope.

**Additional context**
I'm currently investigating various issues regarding memory leaks and handle leaks.
I will try to implement a very simplified version of an MPV wrapper in c#. At least to get more familiar with the libmpv. Which is not very well documented in my opinion.
Hopefully I can find a solution for the race condition and the handle leak. If it isn't an issue in the libmpv itself. Not sure how stable the stable releases of libmpv actually are.

As a side note: I'm trying to read the duration of a couple hundred thousand videos using the above code. I use the mpv.FileLoaded event to know when the duration property is accessable and then I try to quit. Which I previously did by existing the loop and disposing the object. But this seems to keep 4 handles open. After getting the duration of 1053 files, I had about 4k handles open, crashing the application.
Now I tried to properly "unload" the video before exiting. But the only way to do something like that is the "quit" command it seems.

**Comment by hudec117** [comment]
Will be released with 1.2.0 in the next week. Will close after release.

**Comment by hudec117** [comment]
@SebastianBecker2 1.2.0 released, going to close this issue.
